### ISSUE TITLE
Dropbox sync on iOS never starts. Errors out with: Request to files/download failed. Retrying with workaround. 

### ISSUE THREAD

**Issue by baldengineer** [laurent22/joplin#10454]
### Operating system

iOS

### Joplin version

12.14.8

### Current behaviour

iOS Joplin stopped synchronizing via Dropbox. It had been syncing as expected for two months. The last synced note appears to be about six weeks ago.

 I am seeing the same behavior on iPhone and iPad. Syncing via Dropbox with the desktop client is still working as expected.

1. Joplin client starts to Syncrynoize. 
2. The Sync icon endlessly spins
3. An error is generated in the log
4. Tapping `Cancel` shows the message `Cancelling...`, but it never clears, and the sync icon never stops spinning.

## Sync Status
```
Attachments
Not downloaded: 0
Downloading: 0
Downloaded and decrypted: 10030
Downloaded and encrypted: 0
Created locally: 24
Error: 0
---
Sync status (synced items / total items)
Note: 3405/3419
```

### Expected behaviour

I expected Joplin to synchronize notes via Dropbox.

### Logs

```
05-21T08:37:08: file-api-driver-dropbox: "Request to files/download failed. Retrying with workaround. Error: ", "TypeError: Network request failed
TypeError: Network request failed
    at anonymous (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:11103:33)
    at apply (native)
    at anonymous (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:8734:26)
    at _callTimer (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:8682:17)
    at callTimers (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:8829:19)
    at apply (native)
    at __callFunction (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:1827:36)
    at anonymous (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:1666:31)
    at __guard (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:1778:15)
    at callFunctionReturnFlushedQueue (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:1665:21)"
05-21T08:35:40: file-api-driver-dropbox: "Request to files/download failed. Retrying with workaround. Error: ", "TypeError: Network request failed
TypeError: Network request failed
    at anonymous (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:11103:33)
    at apply (native)
    at anonymous (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:8734:26)
    at _callTimer (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:8682:17)
    at callTimers (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:8829:19)
    at apply (native)
    at __callFunction (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:1827:36)
    at anonymous (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:1666:31)
    at __guard (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:1778:15)
    at callFunctionReturnFlushedQueue (/Users/laurent/Library/Developer/Xcode/DerivedData/Joplin-diwyqlzmvthyrbaopjjmecdqvnbe/Build/Intermediates.noindex/ArchiveIntermediates/Joplin/BuildProductsPath/Release-iphoneos/main.jsbundle:1665:21)"
```

**Comment by personalizedrefrigerator** [comment]
Thank you for reporting this!

I'm linking to several related issues, discussions, and commits:
- https://github.com/laurent22/joplin/issues/10396
   - Original workaround (iOS 12.14.8): https://github.com/laurent22/joplin/commit/3042e615ac740507c349c3a3ffaf14c96f331708
   - Updated (unreleased) workaround: https://github.com/laurent22/joplin/commit/3312bd27c90d54309ddc070bed5012edf50c27ad
- [Post on the Joplin Forum](https://discourse.joplinapp.org/t/ios-sync-broke-due-to-network-request-failing-with-dropbox/37915/24)
- [Post about this issue on the Dropbox forum](https://www.dropboxforum.com/t5/Dropbox-API-Support-Feedback/GET-request-to-files-download-fails-with-quot-The-network/td-p/768270/highlight/true/page/2)

**Comment by baldengineer** [comment]
>     * [iOS,Android: Dropbox sync broken #10396](https://github.com/laurent22/joplin/issues/10396)

Thanks for linking to that issue. I didn't find it when I did my search. (I was probably focused on open issues.) It seems related, and the timeline aligns.

Interestingly, my iPhone client started syncing yesterday, but the iPad client still does not. As far as I can tell, I have not changed anything since submitting the report. I just opened Joplin and noticed notes from my Desktop were suddenly popping in.

Also, now on the iPhone client, when I go to `Sync Status`, there is an entry for `Folders` with all my notebooks. That was not there before (and it still is not showing in the iPad client.)

**Comment by baldengineer** [comment]
My iPad client is still unable to sync via Dropbox. I'm not clear from the provided issues if a version with another fix is coming or if it was assumed that 12.14.8 fixed it.

**Comment by personalizedrefrigerator** [comment]
> My iPad client is still unable to sync via Dropbox. I'm not clear from the provided issues if a version with another fix is coming or if it was assumed that 12.14.8 fixed it.

Another fix should be coming:
- There were changes in 12.14.8 that intended to fix the issue (see [this pull request](https://github.com/laurent22/joplin/pull/10411)).
- [This change](https://github.com/laurent22/joplin/commit/3312bd27c90d54309ddc070bed5012edf50c27ad) attempts to fix the issue in a different way. It should be in Joplin 13.0.3 ([available via TestFlight](https://discourse.joplinapp.org/t/a-new-ios-beta-version-is-available/32361)).

**Comment by fesilvajacquier** [comment]
Hey all!

Got here because I had this error:

```
07-02T15:40:29: file-api-driver-dropbox: "Request to files/download failed. Retrying with workaround. Error: ", "TypeError: Network request failed
TypeError: Network request failed
```

Installing (or was it updating) Joplin via TestFlight solved the issue. Now it is syncing.

**Comment by baldengineer** [comment]
Originally, I had trouble with an iPhone and an iPad. The iPhone mysteriously started syncing on 12.14.8 without me changing anything. The iPad, however, did not work until I updated it to 13.0.6. 

So far, 13.0.6 is syncing on both devices. Since a fix was implemented for 13.0.6 and Joplin+Dropbox sync is working on both of my devices, I'll close this issue. 

Thanks for the continued work on the apps.

**Comment by ghost** [comment]
My colleague is having this issue with:

ios 18.0.1
joplin mobile 13.1.7 prod ios
dropbox: 404.2
it stopped working between nov 5 and nov 19, several notes on desktop created in this interval were not synced.
there was a joplin update two weeks ago, which matches the reported interval.

When in the dropbox app, the md files are present and properly synced between mobile and desktop.

Could that be a regression in the last ios release?
Is nextcloud a possible workaround?
